src/api-extension/api-extension.controller.ts

```diff
@@ -25,6 +26,56 @@ export class ApiExtensionController {
     private readonly logger: Logger,
   ) {}
 
+  async handleRequestCart(cart: Cart, responseExpress: Response) {
+    let response;
+    try {
+      response =
+        await this.apiExtensionService.validatePromotionsAndBuildCartActions(
```

Comment:
I suppose we rather want to divide this method into two. ValidatePromtions() and BuildCardActions(). We should avoid making methods with and or because it break Single-responsibility principle